tests/integration/test_select.py

```diff
@@ -226,14 +228,16 @@ def test_case_when_no_else(c):
     assert_eq(actual_df, expected_df, check_dtype=False)
 
 
-def test_singular_column_projection_simple(c):
```

Comment:
Feel like the changes made to this test are significant enough to motivate it being its own test? Unless the purpose of the original singular column projection test was to check for this optimization and it wasn't doing that before.